pallets/order-book/src/cache_data_layer.rs

```diff
@@ -439,6 +460,16 @@ impl<T: Config> DataLayer<T> for CacheDataLayer<T> {
         self.asks.get(order_book_id, price).cloned()
     }
 
+    fn is_asks_full(
```

Comment:
you need to add tests for all new methods in `tests/data_layer.rs`

Reply:
the same for `storage_data_layer.rs`

Reply:
done

Reply:
btw tests for length exposed a nasty bug, much thanks :)